pkg/core/storage/store.go

```diff
@@ -49,6 +49,21 @@ const (
 // when a certain key is not found.
 var ErrKeyNotFound = errors.New("key not found")
 
+// SeekFlag denotes options for Store.Seek operation.
+type SeekFlag byte
+
+// Possible Seek options.
+const (
+	// SeekForwards is a default Store.Seek behaviour: keys are being iterated in an
+	// ascending way starting from the specified key (or the nearest larger key if no
+	// matching key was found).
+	SeekForwards SeekFlag = 1 << iota
+	// SeekBackwards denotes reverse Seek order: keys are being iterated in a
+	// descending way starting from the specified key (or the nearest smaller key if
+	// no matching key was found).
+	SeekBackwards
```

Comment:
Use boolean? I also like LevelDB's `util.Range`, something like `storage.SeekRange{Prefix: p, Start: s, Backwards: true}` might be more convenient to use with sane defaults (most of the time just prefix) than a number of function parameters.

Reply:
The second option is more preferable to me, it looks nice, fixed. But we have https://github.com/nspcc-dev/neo-go/pull/2316#discussion_r776186547, so @fyrchik, do you like  `storage.SeekRange`?

Reply:
`SeekRange` looks ok, I was against multiple boolean flags in function signature. The only thing that bothers me is that `Prefix` and `Start` can conflict with each other but it seems unavoidable and is reflected in documentation.

Reply:
> Prefix and Start can conflict with each other

The possible solution for this is to make `Start` do not be prefixed by `Prefix` and contain only unique part after prefix to start seeking from, e.g. if we're seeking over the set of `[11, 12, 13, 21]` then to get `[12, 13]` we need to specify `Prefix=1` and `Start`=2 instead of `Prefix=1` `Start`=12.

Similar solution is used for Billet traversing process and `findstate` RPC implementation.